EventFilter/L1TRawToDigi/plugins/implementations_stage2/ZDCUnpacker.cc

```diff
@@ -0,0 +1,71 @@
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "EventFilter/L1TRawToDigi/plugins/UnpackerFactory.h"
+
+#include "L1Trigger/L1TCalorimeter/interface/CaloTools.h"
+
+#include "L1TObjectCollections.h"
+
+// #include "L1TStage2Layer2Constants.h"
+#include "ZDCUnpacker.h"
+#include "GTSetup.h"
+
+namespace l1t {
+  namespace stage2 {
+    ZDCUnpacker::ZDCUnpacker() : EtSumZDCCopy_(0) {}
+
+    bool ZDCUnpacker::unpack(const Block& block, UnpackerCollections* coll) {
+      using namespace l1t::stage2;
+      uint32_t zdc_mask = 0x3FF;
+      LogDebug("L1T") << "Block ID  = " << block.header().getID() << " size = " << block.header().getSize();
+
+      int nBX =
+          int(ceil(block.header().getSize() / zdc::nOutputFramePerBX));  // Since there 6 frames per demux output event
+      // expect the first four frames to be the first 4 EtSum objects reported per event (see CMS IN-2013/005)
+
+      // Find the central, first and last BXs
+      int firstBX = -(ceil((double)nBX / 2.) - 1);
+      int lastBX;
+      if (nBX % 2 == 0) {
+        lastBX = ceil((double)nBX / 2.);
+      } else {
+        lastBX = ceil((double)nBX / 2.) - 1;
+      }

```

Comment:
In fact, this whole block can be simplified to
```suggestion
      int lastBX = nBX / 2;
```

Reply:
I can apply these changes but these lines were copied from the etsumunpacker (https://github.com/matt2275/cmssw/blob/ZDCUnpacker/EventFilter/L1TRawToDigi/plugins/implementations_stage2/EtSumUnpacker.cc#L24).

Reply:
Please fix the new code in this PR.
A separate PR can be made to fix the other instances.